app/scripts/lib/migrator/index.js

```diff
@@ -13,10 +13,16 @@ class Migrator {
   // run all pending migrations on meta in place
   migrateData (versionedData = this.generateInitialState()) {
     const remaining = this.migrations.filter(migrationIsPending)
+    if (remaining.length === 0) return versionedData
+
+    const migrations = remaining.map((migration, i) => {
+      if (i === 0) return this.runMigration.bind(this, migration, versionedData)
+      return this.runMigration.bind(this, migration)
+    })
 
     return (
-      asyncQ.eachSeries(remaining, (migration) => this.runMigration(versionedData, migration))
-      .then(() => versionedData)
+      asyncQ.waterfall(migrations)
+      .then((migratedData) => Promise.resolve(migratedData))
```

Comment:
`.then((migratedData) => Promise.resolve(migratedData))`
this can be removed - doesnt do anything